magnet/src/main/java/magnet/Scope.java

```diff
@@ -122,4 +123,6 @@
     /** Disposes this and all children scopes. Notifies instances with {@link Instance#disposer()}. */
     void dispose();
 
+    /** Visits all instances and child scopes of given depth.*/
+    void accept(ScopeVisitor visitor, int depth);
```

Comment:
Nit: What if one does not want to restrict the depth? Use `Integer.MAX_VALUE`? Might be worthwhile to document here.

Reply:
Fixed.